tests/preprocessing/test_warm_up_module.py

```diff
@@ -0,0 +1,161 @@
+    for model_value, pretrained_model_value in zip(
+        model.state_dict().values(), pretrained_model.state_dict().values()
+    ):
+        assert (model_value == pretrained_model_value).all()
+
+    # Check if the weights are different from previous model after loading
+    for old_model_value, model_value in zip(old_model.state_dict().values(), model.state_dict().values()):
+        assert (old_model_value != model_value).any()
+
+
+def test_warm_up_module_loading_different_models_without_mapping() -> None:
+    pretrained_model = SmallCnn()
+    model = ToyConvNet()
+    old_model = copy.deepcopy(model)
+    warmupmodule = WarmedUpModule(pretrained_model=pretrained_model)
+    warmupmodule.load_from_pretrained(model)
+
+    # Check if the weights are same with previous model as loading should not have any effect
+    for old_model_value, model_value in zip(old_model.state_dict().values(), model.state_dict().values()):
+        assert (old_model_value == model_value).any()
+
+
+def test_partial_warm_up_module_loading_different_models_without_mapping() -> None:
+    pretrained_model = ToyConvNet_2()
+    model = ToyConvNet()
+    old_model = copy.deepcopy(model)
+    warmupmodule = WarmedUpModule(pretrained_model=pretrained_model)
+    warmupmodule.load_from_pretrained(model)
+
+    # Check if the weights with same size are loaded from pretrained model and if the weights with different size
+    # are same with previous model as loading should not have any effect
+    for key in model.state_dict().keys():
+        if key in ["conv1.weight", "conv1.bias"]:
+            assert (model.state_dict()[key] == pretrained_model.state_dict()[key]).all()
+        else:
+            assert (model.state_dict()[key] == old_model.state_dict()[key]).all()
+
+
+def test_partial_warm_up_module_loading_similar_models_with_mapping() -> None:
+    pretrained_model = ToyConvNet()
+    model = ToyConvNet()
+    old_model = copy.deepcopy(model)
+    warmupmodule = WarmedUpModule(pretrained_model=pretrained_model)
+    warmupmodule.weights_mapping_dict = {"conv1": "conv1", "conv2": "conv2"}
+    warmupmodule.load_from_pretrained(model)
+
+    # Check if only the weights in mapping are loaded from pretrained model and if the weights not in mapping
+    # are same with previous model as loading should not have any effect
+    for key in model.state_dict().keys():
+        if key in ["conv1.weight", "conv1.bias", "conv2.weight", "conv2.bias"]:
+            assert (model.state_dict()[key] == pretrained_model.state_dict()[key]).all()
+        else:
+            assert (model.state_dict()[key] == old_model.state_dict()[key]).all()
+
+
+def test_global_warm_up_module_loading_fenda_model_with_mapping() -> None:
+    pretrained_model = SmallCnn()
+    model = FendaModel(FeatureCnn(), FeatureCnn(), FendaHeadCnn())
+    old_model = copy.deepcopy(model)
+    print(pretrained_model.state_dict().keys())
```

Comment:
I'd say either drop these print statements or turn them into logs?

Reply:
Oops they are extra. Should have been omitted.